mobile/src/main/java/org/openhab/habdroid/ui/OpenHABAboutActivity.java

```diff
@@ -1,27 +1,45 @@
 package org.openhab.habdroid.ui;
 
 import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
 import android.os.Bundle;
-import android.support.design.widget.TabLayout;
-import android.support.v4.app.Fragment;
-import android.support.v4.app.FragmentManager;
-import android.support.v4.app.FragmentPagerAdapter;
-import android.support.v4.view.PagerAdapter;
-import android.support.v4.view.ViewPager;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
+import android.text.TextUtils;
 import android.util.Log;
-import android.view.LayoutInflater;
 import android.view.MenuItem;
-import android.view.View;
-import android.view.ViewGroup;
 
+import com.danielstone.materialaboutlibrary.MaterialAboutFragment;
+import com.danielstone.materialaboutlibrary.items.MaterialAboutActionItem;
+import com.danielstone.materialaboutlibrary.items.MaterialAboutItemOnClickAction;
+import com.danielstone.materialaboutlibrary.items.MaterialAboutTitleItem;
+import com.danielstone.materialaboutlibrary.model.MaterialAboutCard;
+import com.danielstone.materialaboutlibrary.model.MaterialAboutList;
 import com.mikepenz.aboutlibraries.LibsBuilder;
 
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openhab.habdroid.BuildConfig;
 import org.openhab.habdroid.R;
+import org.openhab.habdroid.core.connection.Connection;
+import org.openhab.habdroid.core.connection.ConnectionFactory;
+import org.openhab.habdroid.core.connection.exception.ConnectionException;
+import org.openhab.habdroid.util.MyHttpClient;
 import org.openhab.habdroid.util.Util;
 
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Locale;
+
+import okhttp3.Call;
+import okhttp3.Headers;
+
 public class OpenHABAboutActivity extends AppCompatActivity {
+    private final static String TAG = OpenHABAboutActivity.class.getSimpleName();
+    private static Connection conn = null;
+    private static int mOpenHABVersion = 0;
```

Comment:
Why make those variables static?

Reply:
AboutMainFragment must be static, so those vars, too.

Reply:
A static Fragment instance is a no-go, as it'll leak the activity. Why must it be static? There must be a different solution...

Reply:
The oh version is in the intent, so it will be passed as fragment argument. But how can I add the connection? I tried to add it to the getIntent().getExtras() bundle and pass that here https://github.com/mueller-ma/openhab.android/blob/430b7eb382a02fea46161e6f0abf47ed2949ed02/mobile/src/main/java/org/openhab/habdroid/ui/OpenHABAboutActivity.java#L57 , but I cannot find the function to add the connection to the bundle. Their are only add functions for int, strings, etc.

Reply:
You can re-fetch it from ConnectionFactory.